mage_ai/cluster_manager/kubernetes/workload_manager.py

```diff
@@ -173,27 +176,80 @@ def create_workload(
         )
         container_config['env'] = env_vars
 
-        containers = [
-            {
-                'name': f'{name}-container',
-                'image': 'mageai/mageai:latest',
-                'ports': [
-                    {
-                        'containerPort': 6789,
-                        'name': 'web'
-                    }
-                ],
-                'volumeMounts': [
-                    {
-                        'name': 'mage-data',
-                        'mountPath': '/home/src'
+        mage_container_config = {
+            'name': f'{name}-container',
+            'image': 'mageai/mageai:latest',
+            'ports': [
+                {
+                    'containerPort': 6789,
+                    'name': 'web'
+                }
+            ],
+            'volumeMounts': [
+                {
+                    'name': 'mage-data',
+                    'mountPath': '/home/src'
+                }
+            ],
+            **container_config,
+        }
+
+        containers = [mage_container_config]
+
+        init_containers = []
+        pre_start_script_path = lifecycle_config.pre_start_script_path
+        if pre_start_script_path:
+            self.configure_pre_start(name, pre_start_script_path, mage_container_config)
+
+            volumes.append(
+                {
+                    'name': 'pre-start-script',
+                    'configMap': {
+                        'name': f'{name}-pre-start',
+                        'items': [
+                            {
+                                'key': 'pre-start.py',
+                                'path': 'pre-start.py',
+                            },
+                            {
+                                'key': 'initial-config.json',
+                                'path': 'initial-config.json',
+                            }
+                        ]
                     }
-                ],
-                **container_config,
-            }
-        ]
+                }
+            )
+
+            init_containers.append(
+                {
+                    'name': f'{name}-pre-start',
+                    'image': 'mageai/pre-start:latest',
```

Comment:
Are you going to publish this docker image?

Reply:
yes i will once the PR is finalized